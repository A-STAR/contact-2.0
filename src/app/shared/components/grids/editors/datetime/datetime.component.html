<div class="datetimepicker">
  <app-date
    [ngModel]="tempValue"
    (ngModelChange)="onDateChange($event)"
  ></app-date>
  <app-time
    [ngModel]="tempValue"
    [displaySeconds]="displaySeconds"
    (ngModelChange)="onTimeChange($event)"
  ></app-time>
  <footer>
    <span (click)="setCurrentTime()">{{ 'default.buttons.now' | translate }}</span>
    <span (click)="onOkClick()">{{ 'default.buttons.ok' | translate }}</span>
  </footer>
</div>
