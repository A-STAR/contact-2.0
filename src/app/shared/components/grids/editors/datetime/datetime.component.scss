@import '../../../../styles/app/variables.scss';

.datetimepicker {
  background-color: #fff;
  box-shadow: 1px 1px 10px 0 rgba(0, 0, 0, 0.2);
  margin: 1px 0;
  padding: 20px;
  user-select: none;
  width: 22em;

  app-date,
  app-time {
    padding-bottom: 16px;
  }

  footer {
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    margin: 0 auto;
    width: 80%;

    span {
      color: $text-light;
      cursor: pointer;
      flex: 0 0 auto;
      transition: color 0.2s ease;
      &:hover {
        color: $link-color;
      }
    }
  }
}
