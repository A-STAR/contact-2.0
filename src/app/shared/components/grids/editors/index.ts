export { DateTimeEditComponent } from './datetime/datetime.component';
export { ValueEditorComponent } from './value/value.component';
