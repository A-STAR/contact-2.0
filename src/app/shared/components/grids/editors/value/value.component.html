<ng-container [ngSwitch]="type">

  <div *ngSwitchCase="1" class="text-right">
    <app-number
      [ngModel]="value"
      (ngModelChange)="onValueChange($event)"
    >
    </app-number>
  </div>

  <div *ngSwitchCase="2">
    <div style="padding: 10px;">
      <app-date
        [ngModel]="value"
        (ngModelChange)="onValueChange($event)"
      ></app-date>
    </div>
  </div>

  <div *ngSwitchCase="3">
    <app-text
      [ngModel]="value"
      (ngModelChange)="onValueChange($event)"
    >
    </app-text>
  </div>

  <div *ngSwitchCase="4">
    <app-select
      [dictCode]="2"
      [ngModel]="value"
      (ngModelChange)="onValueChange($event)"
    ></app-select>
  </div>

  <div *ngSwitchCase="5" class="text-right">
    <app-number
      [ngModel]="value"
      (ngModelChange)="onValueChange($event)"
    >
    </app-number>
  </div>

  <div *ngSwitchCase="6">
    <app-select
      [dictCode]="dictCode"
      [lookupKey]="lookupKey"
      [ngModel]="value"
      (ngModelChange)="onValueChange($event)"
    ></app-select>
  </div>

  <div *ngSwitchCase="7">
    <app-grid-datetime-edit
      [ngModel]="value"
      (ngModelChange)="onValueChange($event)"
    >
    </app-grid-datetime-edit>
  </div>

</ng-container>
