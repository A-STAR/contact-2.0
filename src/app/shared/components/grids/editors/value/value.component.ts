import { Component } from '@angular/core';
import { ICellEditorParams } from 'ag-grid/main';
import { ICellEditorAngularComp } from 'ag-grid-angular';

import { TYPE_CODES } from '@app/core/utils';

@Component({
  selector: 'app-grid-value-edit',
  templateUrl: './value.component.html',
})
export class ValueEditorComponent implements ICellEditorAngularComp {
  private params: ICellEditorParams;

  value: any;
  dictCode: number;

  refresh(): boolean {
    return false;
  }

  get type(): number {
    const { node, valueTypeKey } = this.params as any;
    return node.data[valueTypeKey] || TYPE_CODES.STRING;
  }

  agInit(params: ICellEditorParams): void {
    const { node, dictCode, value } = params as any;
    this.params = params;
    this.value = value;
    this.dictCode = typeof dictCode === 'function' ? dictCode(node.data) : dictCode;
  }

  getValue(): any {
    return this.value;
  }

  isPopup(): boolean {
    return true;
  }

  onValueChange(value: any): void {
    this.value = value;
  }
}
