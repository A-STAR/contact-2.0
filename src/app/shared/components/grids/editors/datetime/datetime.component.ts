import { Component, forwardRef, Input, ChangeDetectorRef } from '@angular/core';
import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';

import { DateTimeService } from '@app/shared/components/form/datetime/datetime.service';

import * as moment from 'moment';

@Component({
  selector: 'app-grid-datetime-edit',
  templateUrl: './datetime.component.html',
  styleUrls: ['./datetime.component.scss'],
  providers: [
    {
      provide: NG_VALUE_ACCESSOR,
      useExisting: forwardRef(() => DateTimeEditComponent),
      multi: true,
    },
  ],
})
export class DateTimeEditComponent implements ControlValueAccessor {
  @Input() required = false;
  @Input() set displaySeconds(displaySeconds: boolean) {
    this._displaySeconds = displaySeconds === undefined ? true : displaySeconds;
    this.dateTimeFormat = this._displaySeconds ? 'L HH:mm:ss' : 'L HH:mm';
  }

  dateTimeFormat = 'L HH:mm:ss';

  value: Date;
  tempValue: Date;

  private _displaySeconds: boolean;

  constructor(
    private cdRef: ChangeDetectorRef,
    private dateTimeService: DateTimeService,
  ) {}

  get displaySeconds(): boolean {
    return this._displaySeconds;
  }

  writeValue(value: Date | string): void {
    if (value) {
      this.value = value instanceof Date
        ? value
        : moment(value).toDate();
    } else {
      this.value = null;
    }
    this.tempValue = this.value;
    this.cdRef.markForCheck();
  }

  registerOnChange(fn: Function): void {
    this.propagateChange = fn;
  }

  registerOnTouched(fn: Function): void {
    this.propagateChange = fn;
  }

  setCurrentTime(): void {
    const value = new Date();
    this.update(value);
  }

  onDateChange(date: Date): void {
    this.tempValue = this.dateTimeService.setDate(this.tempValue, date);
  }

  onTimeChange(time: Date): void {
    this.tempValue = this.dateTimeService.setTime(this.tempValue, time);
  }

  onOkClick(): void {
    this.update(this.tempValue);
  }

  private update(value: Date): void {
    this.value = value;
    this.tempValue = value;
    this.propagateChange(value);
    this.cdRef.markForCheck();
  }

  private propagateChange: Function = () => {};

}
