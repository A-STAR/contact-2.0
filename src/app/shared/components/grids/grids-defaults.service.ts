import { Injectable } from '@angular/core';

import { GridApi, ColumnApi } from 'ag-grid';
import { IGridColumn, IGridLocalSettings } from '@app/shared/components/grids/grids.interface';

import { SettingsService } from '@app/core/settings/settings.service';

@Injectable()
export class GridsDefaultsService {

  private defaultSettings: IGridLocalSettings;
  private persistenceKey: string;

  constructor(
    private settingsService: SettingsService,
  ) { }

  save<T>(columns: IGridColumn<T>[], persistenceKey: string): void {
    this.defaultSettings = {
      columns: columns
        .filter(col => col.prop)
        .map(column => ({
            width: column.maxWidth || column.minWidth || 200,
            isVisible: column.isGroup,
            colId: column.prop,
          })
        ),
      sortModel: [],
    };
    this.persistenceKey = persistenceKey;
  }

  reset(gridApi: GridApi, columnApi: ColumnApi): void {
    columnApi.setColumnState(this.defaultSettings.columns);
    gridApi.setSortModel(this.defaultSettings.sortModel);
    this.setLocalSettings(this.persistenceKey, this.defaultSettings);
  }

  private setLocalSettings(key: string, settings: IGridLocalSettings): void {
    this.settingsService.set(key, settings);
  }

}
