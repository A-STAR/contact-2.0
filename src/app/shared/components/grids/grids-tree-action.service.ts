import { Injectable } from '@angular/core';
import { ColDef } from 'ag-grid';
import { TranslateService } from '@ngx-translate/core';

import { IGridColumn, IGridTreePath } from '@app/shared/components/grids/grids.interface';

import { GridsDefaultsService } from '@app/shared/components/grids/grids-defaults.service';
import { GridsService } from './grids.service';
import { SettingsService } from '@app/core/settings/settings.service';
import { UserDictionariesService } from '@app/core/user/dictionaries/user-dictionaries.service';

import { StateTree } from '@app/core/utils/state-tree';

@Injectable()
export class GridsTreeActionService extends GridsService {
  private actionKeys: string[] = [];
  private stateTree: StateTree;

  constructor(
    settingsService: SettingsService,
    translateService: TranslateService,
    userDictionariesService: UserDictionariesService,
  ) {
    super(
      settingsService,
      translateService,
      userDictionariesService
    );
    this.stateTree = new StateTree({
      transformState: (prev, next) => prev + next,
      dataToState: this.dataToState.bind(this)
    });
   }

  /**
  * @override
  */
   convertColumnsToColDefs<T>(columns: IGridColumn<T>[], persistenceKey: string, defaults?: GridsDefaultsService): ColDef[] {
     this.actionKeys = columns.filter(c => c.hasAction).map(column => column.prop);
     const colDefs = super.convertColumnsToColDefs(columns, persistenceKey, defaults);
      colDefs
        .filter(col => this.actionKeys.includes(col.colId))
        .forEach(c => c.cellRendererParams = {
          ...c.cellRendererParams,
          stateTree: this.stateTree
        });
      return colDefs;
  }
   /**
    * @override
    * @param data
    * @param prop
    * @param parent
    */
   convertTreeData<T>(data: T[], prop: string = 'name', parent?: IGridTreePath): (T & IGridTreePath)[] {

    return data.reduce((acc, item: T & IGridTreePath) => {
      item.path = parent ? [...parent.path, item[prop]] : [item[prop]];

      this.stateTree.addNode(item.path, this.dataToState(item));

      return [
        ...acc,
        item,
        ...(item.children && item.children.length ? this.convertTreeData(item.children, prop, item) : [])
      ];
    }, [])
    // remove children property
    .map(result => {
      const { children, ...rest } = result;
      return rest;
    });
  }

  private dataToState(row: any): number {
    return this.actionKeys.reduce((acc: number, key) => acc + Number(row[key]), 0);
  }

}
