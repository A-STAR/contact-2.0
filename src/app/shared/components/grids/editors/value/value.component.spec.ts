import { async as Async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { TranslateLoader, TranslateModule, TranslateService } from '@ngx-translate/core';
import { LangChangeEvent } from '@ngx-translate/core';
import { Observable } from 'rxjs/Observable';
import { PipeTransform, Pipe } from '@angular/core';
import { of } from 'rxjs/observable/of';

import { IOption } from '@app/core/converter/value-converter.interface';
import { IUserTerm } from '@app/core/user/dictionaries/user-dictionaries.interface';

import { DateTimeModule } from '@app/shared/components/form/datetime/datetime.module';
import { InputModule } from '@app/shared/components/form/input/input.module';
import { MomentModule } from '@app/shared/pipes/moment/moment.module';
import { SingleSelectModule } from '@app/shared/components/form/select/single/single-select.module';

import { LookupService } from '@app/core/lookup/lookup.service';
import { UserDictionariesService } from '@app/core/user/dictionaries/user-dictionaries.service';
import { UserPermissionsService } from '@app/core/user/permissions/user-permissions.service';

import { SortOptionsPipe } from '@app/shared/components/form/select/select.pipe';

import { DateTimeEditComponent } from '@app/shared/components/grids/editors';
import { ValueEditorComponent } from './value.component';

class TranslateLoaderMock {
  getTranslation(): Observable<any> {
    return of([]);
  }
}

class TranslateServiceMock {
  defaultLang = 'ru';
  currentLang = 'ru';

  get(): Observable<string | any> {
    return of('Истина');
  }

  onLangChange: Observable<LangChangeEvent> = of(
    {
      lang: 'ru',
      translations: {
        default: {
          boolean: {
            TRUE: 'Истина',
            FALSE: 'Ложь'
          }
        }
      },
    }
  );
}

@Pipe({ name: 'translate' })
class TranslatePipeMock implements PipeTransform {
  transform(): any {
    return '';
  }
}
@Pipe({ name: 'sortOptionsPipe' })
export class SortOptionsPipeMock implements PipeTransform {

  transform(options: any[]): any[] {
    return options;
  }
}

class UserDictionariesServiceMock {
  getDictionary(): Observable<IUserTerm[]> {
    const terms = Array(10).fill(null).map((_, i) => i + 1).map(code => ({
      code,
      name: `Term ${code}`,
      isClosed: false,
      parentCode: null,
      parentDictTerm: null,
    }));
    return of(terms);
  }

  getDictionaryAsOptions(id: number): Observable<IOption[]> {
    let payload;
    switch (id) {
      case 2:
        payload = [
          {
            label: 'Истина',
            value: 1
          },
          {
            label: 'Ложь',
            value: 0
          }
        ];
        break;
      default:
        payload = [
          {
            label: 'One',
            value: 1
          },
          {
            label: 'Two',
            value: 2
          },
          {
            label: 'Three',
            value: 3
          }
        ];
    }
    return of(payload);
  }
}

class LookupServiceMock {
  lookupAsOptions(): Observable<Array<IOption>> {
    return of(
      [
        {
          label: 'One',
          value: 1
        }
      ]
    );
  }
}

class UserPermissionsServiceMock {
  get(): Observable<any[]> {
    return of([]);
  }
}

describe('ValueEditorComponent', () => {
  let fixture: ComponentFixture<ValueEditorComponent>;

  beforeEach(Async(() => {
    TestBed
      .configureTestingModule({
        imports: [
          FormsModule,
          DateTimeModule,
          InputModule,
          MomentModule,
          SingleSelectModule,
          TranslateModule.forRoot({
            loader: {
              provide: TranslateLoader,
              useClass: TranslateLoaderMock,
            },
          }),
        ],
        providers: [
          {
            provide: TranslateService,
            useClass: TranslateServiceMock
          },
          {
            provide: UserDictionariesService,
            useClass: UserDictionariesServiceMock
          },
          {
            provide: LookupService,
            useClass: LookupServiceMock
          },
          {
            provide: SortOptionsPipe,
            useClass: SortOptionsPipeMock
          },
          {
            provide: UserPermissionsService,
            useClass: UserPermissionsServiceMock,
          },
        ],
        declarations: [
          DateTimeEditComponent,
          SortOptionsPipeMock,
          TranslatePipeMock,
          ValueEditorComponent,
        ]
      })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ValueEditorComponent);
  });

  it('should render value editor with typeCode 1 (number)', () => {
    fixture.componentInstance.agInit({
      value: 0,
      node: {
        data: {
          typeCode: 1
        }
      },
      valueTypeKey: 'typeCode',
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  // it('should render value editor with typeCode 2 (datepicker)', () => {
  //   fixture.componentInstance.agInit({
  //     value: '2017-06-20T21:00:00Z',
  //     node: {
  //       data: {
  //         typeCode: 2
  //       }
  //     },
  //     valueTypeKey: 'typeCode',
  //   } as any);
  //   fixture.detectChanges();
  //   expect(fixture.nativeElement).toMatchSnapshot();
  // });

  it('should render value renderer with typeCode 3 (string)', () => {
    fixture.componentInstance.agInit({
      value: 'value',
      node: {
        data: {
          typeCode: 3
        }
      },
      valueTypeKey: 'typeCode',
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render value editor with typeCode 4 (boolean)', () => {
    fixture.componentInstance.agInit({
      value: true,
      node: {
        data: {
          typeCode: 4
        }
      },
      valueTypeKey: 'typeCode',
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render value editor with typeCode 5 (float)', () => {
    fixture.componentInstance.agInit({
      value: 3.14,
      node: {
        data: {
          typeCode: 5
        }
      },
      valueTypeKey: 'typeCode',
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render value editor with typeCode 6 (dictionary) by dictCode', () => {
    fixture.componentInstance.agInit({
      value: 33,
      node: {
        data: {
          typeCode: 6
        }
      },
      dictCode: 33,
      valueTypeKey: 'typeCode',
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render value editor with typeCode 6 (dictionary) by dictCode callback', () => {
    fixture.componentInstance.agInit({
      value: 33,
      node: {
        data: {
          typeCode: 6,
          dictCode: 33
        }
      },
      dictCode: item => item.dictCode,
      valueTypeKey: 'typeCode',
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  it('should render value editor with typeCode 6 (dictionary) by lookupKey', () => {
    fixture.componentInstance.agInit({
      value: 33,
      node: {
        data: {
          typeCode: 6,
          lookupKey: 'users'
        }
      },
      valueTypeKey: 'typeCode',
    } as any);
    fixture.detectChanges();
    expect(fixture.nativeElement).toMatchSnapshot();
  });

  // it('should render value editor with typeCode 7 (datetimepicker)', () => {
  //   fixture.componentInstance.agInit({
  //     value: '2017-06-20T21:00:00Z',
  //     node: {
  //       data: {
  //         typeCode: 7
  //       }
  //     },
  //     valueTypeKey: 'typeCode',
  //   } as any);
  //   fixture.detectChanges();
  //   expect(fixture.nativeElement).toMatchSnapshot();
  // });

});
