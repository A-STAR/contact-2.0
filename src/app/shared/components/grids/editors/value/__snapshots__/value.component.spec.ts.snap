// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueEditorComponent should render value editor with typeCode 1 (number) 1`] = `
<div
  id="root0"
  ng-version="5.2.4"
>
  
  

  
  
  <div
    class="text-right"
  >
    
    
    <app-number
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-model="0"
    >
      <label
        class="m0"
      >
        
  
        
        
  
        <div
          class="input-container"
        >
          
    
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            ng-reflect-is-disabled="false"
            type="text"
          />
          
    
          <i
            class="fa fa-angle-up"
          />
          
    
          <i
            class="fa fa-angle-down"
          />
          
  
        </div>
        
  
        <div>
          
    
          
          
  
        </div>
        

      </label>
      

    </app-number>
    
  
  </div>
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  


  

</div>
`;

exports[`ValueEditorComponent should render value editor with typeCode 4 (boolean) 1`] = `
<div
  id="root2"
  ng-version="5.2.4"
>
  
  

  
  
  

  
  
  

  
  
  

  
  
  <div>
    
    
    <app-select
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-dict-code="2"
      ng-reflect-model="true"
    >
      

      <label
        class="m0"
      >
        

  
        
        

  
        <div
          class="select-container dropdown open"
        >
          

    
          <input
            autocomplete="off"
            class="form-control select-search ng-untouched ng-pristine ng-valid"
            ng-reflect-model=""
            placeholder=""
            type="text"
          />
          
    
          
          <i
            class="co co-back select-caret"
            ng-reflect-klass="co co-back select-caret"
            ng-reflect-ng-class=""
          />
          
    
          
          

    
          
          
  
        </div>
        

  
        <div>
          
    
          
          
  
        </div>
        


      </label>
      

    </app-select>
    
  
  </div>
  

  
  
  

  
  
  

  
  
  


  

</div>
`;

exports[`ValueEditorComponent should render value editor with typeCode 5 (float) 1`] = `
<div
  id="root3"
  ng-version="5.2.4"
>
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  <div
    class="text-right"
  >
    
    
    <app-number
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-model="3.14"
    >
      <label
        class="m0"
      >
        
  
        
        
  
        <div
          class="input-container"
        >
          
    
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            ng-reflect-is-disabled="false"
            type="text"
          />
          
    
          <i
            class="fa fa-angle-up"
          />
          
    
          <i
            class="fa fa-angle-down"
          />
          
  
        </div>
        
  
        <div>
          
    
          
          
  
        </div>
        

      </label>
      

    </app-number>
    
  
  </div>
  

  
  
  

  
  
  


  

</div>
`;

exports[`ValueEditorComponent should render value editor with typeCode 6 (dictionary) by dictCode 1`] = `
<div
  id="root4"
  ng-version="5.2.4"
>
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  <div>
    
    
    <app-select
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-dict-code="33"
      ng-reflect-model="33"
    >
      

      <label
        class="m0"
      >
        

  
        
        

  
        <div
          class="select-container dropdown open"
        >
          

    
          <input
            autocomplete="off"
            class="form-control select-search ng-untouched ng-pristine ng-valid"
            ng-reflect-model=""
            placeholder=""
            type="text"
          />
          
    
          
          <i
            class="co co-back select-caret"
            ng-reflect-klass="co co-back select-caret"
            ng-reflect-ng-class=""
          />
          
    
          
          

    
          
          
  
        </div>
        

  
        <div>
          
    
          
          
  
        </div>
        


      </label>
      

    </app-select>
    
  
  </div>
  

  
  
  


  

</div>
`;

exports[`ValueEditorComponent should render value editor with typeCode 6 (dictionary) by dictCode callback 1`] = `
<div
  id="root5"
  ng-version="5.2.4"
>
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  <div>
    
    
    <app-select
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-dict-code="33"
      ng-reflect-model="33"
    >
      

      <label
        class="m0"
      >
        

  
        
        

  
        <div
          class="select-container dropdown open"
        >
          

    
          <input
            autocomplete="off"
            class="form-control select-search ng-untouched ng-pristine ng-valid"
            ng-reflect-model=""
            placeholder=""
            type="text"
          />
          
    
          
          <i
            class="co co-back select-caret"
            ng-reflect-klass="co co-back select-caret"
            ng-reflect-ng-class=""
          />
          
    
          
          

    
          
          
  
        </div>
        

  
        <div>
          
    
          
          
  
        </div>
        


      </label>
      

    </app-select>
    
  
  </div>
  

  
  
  


  

</div>
`;

exports[`ValueEditorComponent should render value editor with typeCode 6 (dictionary) by lookupKey 1`] = `
<div
  id="root6"
  ng-version="5.2.4"
>
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  

  
  
  <div>
    
    
    <app-select
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-model="33"
    >
      

      <label
        class="m0"
      >
        

  
        
        

  
        <div
          class="select-container dropdown open"
        >
          

    
          <input
            autocomplete="off"
            class="form-control select-search ng-untouched ng-pristine ng-valid"
            ng-reflect-model=""
            placeholder=""
            type="text"
          />
          
    
          
          <i
            class="co co-back select-caret"
            ng-reflect-klass="co co-back select-caret"
            ng-reflect-ng-class=""
          />
          
    
          
          

    
          
          
  
        </div>
        

  
        <div>
          
    
          
          
  
        </div>
        


      </label>
      

    </app-select>
    
  
  </div>
  

  
  
  


  

</div>
`;

exports[`ValueEditorComponent should render value renderer with typeCode 3 (string) 1`] = `
<div
  id="root1"
  ng-version="5.2.4"
>
  
  

  
  
  

  
  
  

  
  
  <div>
    
    
    <app-text
      class="ng-untouched ng-pristine ng-valid"
      ng-reflect-model="value"
    >
      <label
        class="m0"
      >
        
  
        
        
  
        <div>
          
    
          <input
            class="form-control ng-untouched ng-pristine ng-valid"
            ng-reflect-is-disabled="false"
            type="text"
          />
          
  
        </div>
        
  
        <div>
          
    
          
          
  
        </div>
        

      </label>
      

    </app-text>
    
  
  </div>
  

  
  
  

  
  
  

  
  
  

  
  
  


  

</div>
`;
